### PR TITLE
TPROD-399 Removed Doctrine deprecations from Campaign mautic:campaigns:rebuild command.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Automated tests included?              | [ x ] 

#### Description:
Remove Doctrine deprecations from Campaign mautic:campaigns:rebuild command.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create Campaign
3. Create contact that does not belong to the campaign.
4. Create contact that belong to the campaign.
5. Run `mautic:campaigns:rebuild` command.

### CODE REVIEW

**Review by mabumusa1 — DISMISSED** [review]
This PR is hard to test on gitpod at the moment given the [issue](https://github.com/mautic/mautic/pull/12166#issuecomment-1498907535) 

I tested it locally with a campaign that was saved in the database already, the command worked as expected

File: app/bundles/CampaignBundle/Membership/MembershipBuilder.php [hunk 1]
```diff
@@ -154,7 +159,8 @@ private function addNewlyQualifiedMembers(int $totalContactsProcessed): int
             $this->manager->addContacts($contactCollection, $this->campaign, false);
 
             // Clear Lead entities from RAM
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Membership/MembershipBuilder.php:156]: I'm not sure that this comment still makes sense. The change will clear all the entities. Not just the Lead entities. @fedys do you know about better replacement than the one that is suggested here?

Comment (fedys) [reply app/bundles/CampaignBundle/Membership/MembershipBuilder.php:156]: I would say
```php
$this->leadRepository->detachEntities(contacts);
```

Comment (fedys) [reply app/bundles/CampaignBundle/Membership/MembershipBuilder.php:156]: But it is not the same as calling `clear()`.

Comment (fedys) [reply app/bundles/CampaignBundle/Membership/MembershipBuilder.php:156]: I checked the internals of Doctrine UoW and the only difference between calling `$this->leadRepository->detachEntities(contacts);` and `$this->leadRepository->clear();` in this particular situation is that the latter triggers `Events::onClear` event. As we have no such listener, there should not be any difference.

File: app/bundles/CampaignBundle/Membership/MembershipBuilder.php [hunk 2]
```diff
@@ -154,7 +159,8 @@ private function addNewlyQualifiedMembers(int $totalContactsProcessed): int
...
-            $this->leadRepository->clear();
+            $this->objectManager->flush();
+            $this->objectManager->clear();
```
Comment (fedys) [inline app/bundles/CampaignBundle/Membership/MembershipBuilder.php]:
```suggestion
            $this->leadRepository->detachEntities(contacts);
```
Let's replace it with detaching entities.

**Review by fedys — CHANGES_REQUESTED** [review]

File: app/bundles/CampaignBundle/Membership/MembershipBuilder.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/CampaignBundle/Membership/MembershipBuilder.php:156]: Please check now

Comment (fedys) [reply app/bundles/CampaignBundle/Membership/MembershipBuilder.php:156]: Looks good. Thanks!

**Review by fedys — APPROVED** [review]

**Review by fedys — APPROVED** [review]
Thanks @biozshock! 👍

**Review by escopecz — APPROVED** [review]
Thank you!

**Pull request merged**